server/routerlicious/packages/routerlicious-base/src/riddler/tenantManager.ts

```diff
@@ -278,4 +285,38 @@ export class TenantManager {
 
         await collection.update({ _id: tenantId }, { disabled: true }, null);
     }
+
+    private encryptAccessInfo(externalStorageData: any): any {
+        const accessInfo = externalStorageData.accessInfo;
```

Comment:
I am assuming the entire accessInfo object needs to be encrypted? Then why not just stringifying it and encode the whole thing?

Reply:
@tanviraumi I was wondering about this actually. Would it be best to stringify and encrypt the entirety of `customData`, then decrypt and parse it on retrieval? That way there would be no need to specifiy which part to encrypt/decrypt, and `customData` could remain generic

Reply:
I am not sure what else is stored inside customData. If its secret (fields we should encrypt) heavy, we can consider encrypting the whole thing. My only concern is the cost and latency for encrypting and decrypting. But if we can consolidate the secret fields into one field, then we should just encrypt the whole field. It still remains generic and anything we will add later will be auto encrypted.

Reply:
@tanviraumi I will change the code to encrypt the entire object instead. so far I think accessInfo is the only field that need to be encrypted.

Reply:
@Noelin I think what Tanvir is suggesting is that, in order to avoid potential latency from encrypting/decrypting the entire object, there should be a designated (optional) property on `customData` called `secrets` that is always encrypted/decrypted. It would be up to the consumer (i.e. caller of the service) to structure the `customData` in a way that all secrets that must be encrypted are properties of `customData.secrets`.

To illustrate, [ITenantCustomData](https://github.com/microsoft/FluidFramework/blob/main/server/routerlicious/packages/services-core/src/tenant.ts#L52-L54) would look like this
```ts
export interface ITenantCustomData {
    /*
     * When present, the `secrets` object is always encrypted for storage and decrypted on retrieval.
     */
    secrets?: {
        [key: string]: any;
    };
    [key: string]: any;
}
```

@tanviraumi did I interpret that correctly?

Reply:
That sounds good. Does it make sense to stage it though? For this PR, Lin can encrypt customData itself and follow it up with the reorganization of secrets in custom data so that we can be more deliberate about what we encrypt and how we group secrets in customdata.

Reply:
@znewton Isn't that what is happening now though? accessInfo mostly contains the secrets. 
![image](https://user-images.githubusercontent.com/448106/102679472-0406f800-4165-11eb-80d6-06ce2531d8b0.png)

Reply:
@znewton : Exactly! Thats what I am looking for. As you already mentioned, it keep it generic for future changes.

But for this PR, I think encrypting accessInfo is good enough I think. Unless we need more fields to encrypt.